fe/fe-core/src/main/java/com/starrocks/connector/ConnectorPartitionTraits.java

```diff
@@ -435,6 +445,53 @@ public Optional<Long> maxPartitionRefreshTs() {
             IcebergTable icebergTable = (IcebergTable) table;
             return icebergTable.getSnapshot().map(Snapshot::timestampMillis);
         }
+
+        @Override
+        public PartitionKey createPartitionKey(List<String> values, List<Column> columns) throws AnalysisException {
+            Preconditions.checkState(values.size() == columns.size(),
+                    "columns size is %s, but values size is %s", columns.size(), values.size());
+
+            IcebergTable icebergTable = (IcebergTable) table;
+            List<PartitionField> partitionFields = Lists.newArrayList();
+            for (Column column : columns) {
+                for (PartitionField field : icebergTable.getNativeTable().spec().fields()) {
+                    String partitionFieldName = icebergTable.getNativeTable().schema().findColumnName(field.sourceId());
+                    if (partitionFieldName.equalsIgnoreCase(column.getName())) {
+                        partitionFields.add(field);
+                    }
+                }
+            }
+            Preconditions.checkState(partitionFields.size() == columns.size(),
+                    "columns size is %s, but partitionFields size is %s", columns.size(), partitionFields.size());
+
+            PartitionKey partitionKey = createEmptyKey();
+
+            // change string value to LiteralExpr,
+            for (int i = 0; i < values.size(); i++) {
+                String rawValue = values.get(i);
+                Column column = columns.get(i);
+                PartitionField field = partitionFields.get(i);
+                LiteralExpr exprValue;
+                // rawValue could be null for delta table
+                if (rawValue == null) {
+                    rawValue = "null";
+                }
+                if (((NullablePartitionKey) partitionKey).nullPartitionValueList().contains(rawValue)) {
+                    partitionKey.setNullPartitionValue(rawValue);
+                    exprValue = NullLiteral.create(column.getType());
+                } else {
+                    if (field.transform().dedupName().equalsIgnoreCase("time")) {
```

Comment:
could you add some comments to explain it ?

Reply:
Done